include/deal.II/matrix_free/evaluation_template_factory.templates.h

```diff
@@ -402,6 +402,53 @@ namespace internal
       fe_degree, n_q_points_1d, n_components, fe_eval, in_array, out_array);
   }
 
+
+
+  template <int dim, typename Number, typename VectorizedArrayType>
+  void
+  FEEvaluationHangingNodesFactory<dim, Number, VectorizedArrayType>::apply(
+    const unsigned int n_components,
+    const unsigned int fe_degree,
+    const FEEvaluationBaseData<dim, Number, false, VectorizedArrayType>
+      &                                                          fe_eval,
+    const bool                                                   transpose,
+    const std::array<unsigned int, VectorizedArrayType::size()> &c_mask,
+    VectorizedArrayType *                                        values)
+  {
+    instantiation_helper_run<
+      1,
+      FEEvaluationImplHangingNodes<dim, VectorizedArrayType, false>>(
+      fe_degree,
+      fe_degree + 1,
+      n_components,
+      fe_eval,
+      transpose,
+      c_mask,
+      values);
+  }
+
+
+
+  template <int dim, typename Number, typename VectorizedArrayType>
+  void
+  FEEvaluationHangingNodesFactory<dim, Number, VectorizedArrayType>::apply(
+    const unsigned int n_components,
+    const unsigned int fe_degree,
+    const FEEvaluationBaseData<dim, Number, true, VectorizedArrayType> &fe_eval,
+    const bool                                                   transpose,
+    const std::array<unsigned int, VectorizedArrayType::size()> &c_mask,
+    VectorizedArrayType *                                        values)
+  {
+    Assert(false, ExcNotImplemented());
+
+    (void)n_components;
+    (void)fe_degree;
+    (void)fe_eval;
+    (void)transpose;
+    (void)c_mask;
+    (void)values;
```

Comment:
> Does this mean that face integrals computed on FE_Q with hanging nodes (e.g. for inhomogeneous Neumann conditions or weak Dirichlet conditions) end up here?

At the moment, this function is not called for faces, since

https://github.com/dealii/dealii/blob/85f67b5422d6cc2596032f6202122c12a8ac3207/include/deal.II/matrix_free/fe_evaluation.h#L5307-L5312

It is more a way to please the compiler.

> But we should be able to use the same code here, shouldn't we? While there could be a case where FE_Q goes into the contiguous unknown section, I think we should be able to rule that part out. It seems the matrix_vector_faces_xx tests miss the part of FE_Q combined with hanging nodes. We do not have to do it in this PR, but it should happen soon. I can do that next week.

I guess the code might work if the face dofs are embedded into the cell dofs. Of course, the mask would need to be updated.